orbit/pkg/swiftdialog/run.go

```diff
@@ -0,0 +1,456 @@
+	commandFile, err := os.CreateTemp("", "swiftDialogCommand")
+	if err != nil {
+		return nil, err
+	}
+
+	jsonBytes, err := json.Marshal(options)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
+
+	if err := commandFile.Chmod(CommandFilePerms); err != nil {
+		commandFile.Close()
+		os.Remove(commandFile.Name())
+		cancel()
+		return nil, err
+	}
+
+	cmd := exec.CommandContext( //nolint:gosec
+		ctx,
+		swiftDialogBin,
+		"--jsonstring", string(jsonBytes),
+		"--commandfile", commandFile.Name(),
+		"--json",
+	)
+
+	outBuf := &bytes.Buffer{}
+	cmd.Stdout = outBuf
+
+	err = cmd.Start()
+	if err != nil {
+		cancel()
+		return nil, err
+	}
+
+	sd := &SwiftDialog{
+		cancel:      cancel,
+		cmd:         cmd,
+		commandFile: commandFile,
+		context:     ctx,
+		done:        make(chan struct{}),
+		output:      outBuf,
+	}
+
+	go func() {
+		if err := cmd.Wait(); err != nil {
+			errExit := &exec.ExitError{}
+			if errors.As(err, &errExit) && strings.Contains(errExit.Error(), "exit status") {
+				sd.exitCode = ExitCode(errExit.ExitCode())
+			} else {
+				sd.exitErr = fmt.Errorf("waiting for swiftDialog: %w", err)
+			}
+		}
+		close(sd.done)
+		cancel()
+	}()
+
+	return sd, nil
+}
+
+func (s *SwiftDialog) finished() {
+	<-s.done
+}
+
+func (s *SwiftDialog) Kill() error {
+	s.cancel()
+	s.finished()
+	if err := s.cleanup(); err != nil {
+		return fmt.Errorf("Close cleaning up after swiftDialog: %w", err)
+	}
+
+	return nil
+}
+
+func (s *SwiftDialog) cleanup() error {
+	s.cancel()
+	cmdFileName := s.commandFile.Name()
+	err := s.commandFile.Close()
+	if err != nil {
+		return fmt.Errorf("closing swiftDialog command file: %w", err)
+	}
+	err = os.Remove(cmdFileName)
+	if err != nil {
+		return fmt.Errorf("removing swiftDialog command file: %w", err)
+	}
+
+	return nil
+}
+
+func (s *SwiftDialog) Wait() (*SwiftDialogExit, error) {
+	s.finished()
+
+	parsed := map[string]any{}
+	if s.output.Len() != 0 {
+		if err := json.Unmarshal(s.output.Bytes(), &parsed); err != nil {
+			return nil, fmt.Errorf("parsing swiftDialog output: %w", err)
+		}
+	}
+
+	if err := s.cleanup(); err != nil {
+		return nil, fmt.Errorf("Wait cleaning up after swiftDialog: %w", err)
+	}
+
+	return &SwiftDialogExit{
+		ExitCode: s.exitCode,
+		Output:   parsed,
+	}, s.exitErr
+}
+
+func (s *SwiftDialog) sendCommand(command, arg string) error {
+	if err := s.context.Err(); err != nil {
+		return fmt.Errorf("could not send command: %w", err)
+	}
+	// For some reason swiftDialog needs us to open and close the file
+	// to detect a new command, just writing to the file doesn't cause
+	// a change
+	commandFile, err := os.OpenFile(s.commandFile.Name(), os.O_APPEND|os.O_WRONLY|os.O_CREATE, CommandFilePerms)
+	if err != nil {
+		return fmt.Errorf("opening command file for writing: %w", err)
+	}
+
+	fullCommand := fmt.Sprintf("%s: %s", command, arg)
+	// fmt.Printf("fullCommand: %v\n", fullCommand)
```

Comment:
nit: can remove I think

Reply:
I was using that for debugging, forgot to remove it